openmindat/geomaterials.py

```diff
@@ -1394,14 +1421,39 @@ def save(self, FILE_NAME = ''):
 
             Example:
                 >>> gir = GeomaterialIdRetriever()
-                >>> gir.save()
+                >>> gir.id(5).save()
         '''
         file_name = FILE_NAME
         
         self.saveto('', file_name)
         
+    def get_dict(self):
+        '''
+        Executes the query to retrieve geomaterial with a corresponding id and returns a dictionary.
+
+        Returns:
+            dictionary.
+
+        Example:
+            >>> gir = GeomaterialIdRetriever()
+            >>> geo5 = gir.id(5).get_dict()
+
+        '''
         
-        #NOT YET WORKING
+        print("Retrieving geomaterials. This may take a while... ")
+       
+        params = self._params
+        end_point = '/'.join([self.end_point, self.sub_endpoint])
+        
+        #clears params for next get statement     
+        self._init_params()
+        
+        ma = mindat_api.MindatApi()
+        return ma.get_mindat_dict(params, end_point)
+        
+        
+        
+        #NOT YET WORKING, check in to see if it returns list vs item
 class GeomaterialDictRetriever:
```

Comment:
Interesting endpoint... It declares that it is retrieving dict objects... Maybe we can keep it as is.